src/ray/object_manager/plasma/stats_collector.cc

```diff
@@ -166,6 +168,27 @@ void ObjectStatsCollector::OnObjectRefDecreased(const LocalObject &obj) {
   }
 }
 
+void ObjectStatsCollector::RecordMetrics() const {
+  ray::stats::STATS_num_objects_by_state.Record(num_objects_spillable_, "Spillable");
+  ray::stats::STATS_num_object_bytes_by_state.Record(num_bytes_spillable_, "Spillable");
+  ray::stats::STATS_num_objects_by_state.Record(num_objects_unsealed_, "Unsealed");
+  ray::stats::STATS_num_object_bytes_by_state.Record(num_bytes_unsealed_, "Unsealed");
+  ray::stats::STATS_num_objects_by_state.Record(num_objects_in_use_, "InUse");
+  ray::stats::STATS_num_object_bytes_by_state.Record(num_bytes_in_use_, "InUse");
+  ray::stats::STATS_num_objects_by_state.Record(num_objects_evictable_, "Evictable");
+  ray::stats::STATS_num_object_bytes_by_state.Record(num_bytes_evictable_, "Evictable");
+
+  ray::stats::STATS_num_objects_by_type.Record(num_objects_created_by_worker_, "Primary");
+  ray::stats::STATS_num_object_bytes_by_type.Record(num_bytes_created_by_worker_,
```

Comment:
We shouldn't have "by type" and "by state". Instead, you can label a single metric with both labels, like

ray::stats::STATS_num_objects.Record(num_bytes_foo, {{"Type", "foo"}, {"State", "bar"}}).

Note you might have to refactor the recording a bit.